paper.tex

```diff
@@ -228,33 +250,127 @@ \subsection*{High-level model specification in YAML}
 \inputminted[fontsize=\scriptsize,linenos,numbersep=5pt]{yaml}{models/IM-resolved.yaml}
 \end{tcolorbox}
 \end{minipage}\\
-\captionof{listing}{An IM model.}
-\label{code:im_model}
+\caption{\label{fig-example}
+Example isolation-with-migration Demes model. (A) The compact non-redundant
+representation expressed using YAML. (B) The same model in the fully-qualified
+form. (C) A visual representation of the model using \texttt{demesdraw}.
+The YAML descriptions in (A) and (B) correspond exactly to a JSON description,
+but are much more concise.
+}
 \end{figure}
 
-\subsection*{Unambiguous low-level representation}
 
-\begin{itemize}
-    \item The Python API, which can read in a YAML file or be built using the
-        Builder
-    \item Demographic model validitation
-    \item Unambiguous, redundant information
-    \item Designed to be programatically read by simulation software backends
-\end{itemize}
+\subsection*{High-level model specification}
+
+The high-level description of Demes models is focused on the efficiency of
+human understanding and the avoidance of errors. We have adopted the widely
+used YAML format~\citep{ben2009yaml} as the recommended means of interchanging
+Demes models (see Figure~\ref{fig-example} for exampels). YAML is a data
+serialisation language with an emphasis on simplicity, and interoperates well
+with JSON (indeed, YAML 1.2 is a superset of JSON). We chose YAML over JSON
+because although JSON is an excellent format for data interchange, it is
+ill-suited for human understanding or manipulation. We also considered other
+data formats such as TOML, but chose YAML because of its greater popularity and
+software support. Since the Demes data model is defined in JSON Schema,
```

Comment:
I think the reason for rejecting TOML was the awkwardness of expressing nested structures, not the level of software support (which is actually quite extensive). Maybe this should also make it clear that we looked at other *declarative* formats. Actually, there aren't so many declarative data formats worth considering!

I looked for alternatives on multiple occasions, and mostly found criticisms of YAML and JSON, for which the solutions were minor incompatible changes to YAML or JSON. One interesting format is [RON](https://github.com/ron-rs/ron), but this has no support outisde Rust, and has some specifics that might map poorly in other languages anyway. But the RON folks have a nice list of possible alternative formats: https://github.com/ron-rs/ron/issues?q=+label%3Aalternative+

Reply:
OK, good point. I've changed it slightly to say that we chose YAML because it has a direct corresondence with JSON (which is definitely a major bonus for interoperability across languages)